src/ucs/datastruct/ptr_array.h

```diff
@@ -149,30 +150,41 @@ void *ucs_ptr_array_replace(ucs_ptr_array_t *ptr_array, unsigned index,
                     (UCS_V_INITIALIZED(_var), 0) : \
                     !__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index])))
 
+
 /**
- * Check if element is free, and calculate how many free elements are ahead.
+ * For-each user function: Calculates how many free elements are ahead.
  *
- * @param [in]     _elem  An element in the ptr array.
- * @param [in/out] iter   Pointer to the foreach iterator, updated to next_idx.
+ * @param [in]  _elem        An element in the ptr array.
  *
- * @return 1 if the element is free and 0 if it's occupied.
+ * @return _size_elem - The number of free elements ahead if free, if not 1.
  */
-static inline int
-__ucs_ptr_array_is_free_and_get_step_size(ucs_ptr_array_elem_t _elem,
-                                          uint32_t *_size_elem)
+static inline uint32_t
+__ucs_ptr_array_for_each_get_step_size(ucs_ptr_array_elem_t _elem)
 {
-    int _is_free = ((uintptr_t)(_elem) & UCS_PTR_ARRAY_FLAG_FREE);
+    int is_free = ((uintptr_t)(_elem) & UCS_PTR_ARRAY_FLAG_FREE);
```

Comment:
`__ucs_ptr_array_is_free()` can be used here

Reply:
Fixed.